### PR TITLE
[App Service] `az functionapp/logicapp create`: Add new `--https-only` parameter

### PR DESCRIPTION
**Pull request by StrawnSC**
**Related command**

**Description**

**Testing Guide**

**History Notes**

{App Service} Upgrade azure-mgmt-web to version 7.0.0
[App Service] `az functionapp/logicapp create`: Add new `--https-only` parameter
[App Service] `az functionapp/webapp create`: Allow vnet integration for basic and elastic premium SKUs

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/requirements.py3.windows.txt
Comment (panchagnula) [inline src/azure-cli/requirements.py3.windows.txt:36]: We shouldn't be updating other SDKs, any reason for this change?

**Comment by panchagnula** [comment]
@StrawnSC please do run all tests locally in live mode with the new SDK change (after running azdev setup) to identify any updates / changes. Thanks!

**Comment by StrawnSC** [comment]
> @StrawnSC please do run all tests locally in live mode with the new SDK change (after running azdev setup) to identify any updates / changes. Thanks!

Yes, I went ahead and pushed all the rerecorded tests. It looks like there are a few that are still failing which I'll need to figure out

File: src/azure-cli/requirements.py3.windows.txt
Comment (StrawnSC) [reply src/azure-cli/requirements.py3.windows.txt:36]: I had to make this change in order for `azdev setup` to run: the new version of `azure-mgmt-web` depends on this new version of `azure-mgmt-core`

Comment (panchagnula) [reply src/azure-cli/requirements.py3.windows.txt:36]: @zhoxing-ms can you comment on this?

Comment (panchagnula) [reply src/azure-cli/requirements.py3.windows.txt:36]: > @zhoxing-ms will I need to test the impact of the `azure-mgmt-core` on other command modules? For what it's worth, the [change log for version 1.3.1 ](https://pypi.org/project/azure-mgmt-core/)has only one entry (so it seems to be a minor update)

@StrawnSC if this is needed, please have the core team handle this change in a separte PR & make this change dependent on that. Our team doesn't own this SDK & hence we shouldn't be making these changes or be responsible for testing & handling issues that might occur due to this update.

**Comment by zhoxing-ms** [comment]
> E               vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ('/home/vsts/work/1/s/src/azure-cli/azure/cli/command_modules/monitor/tests/latest/recordings/test_metric_alert_v2_scenario.yaml') in your current record mode ('once').
E               No match for the request (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v2000001/providers/Microsoft.Web/serverfarms/plan1?api-version=2022-03-01>](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v2000001/providers/Microsoft.Web/serverfarms/plan1?api-version=2022-03-01%3E)) was found.
E               Found 1 similar requests with 1 different matcher(s) :
E               
E               1 - (<Request (PUT) [https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v2000001/providers/Microsoft.Web/serverfarms/plan1?api-version=2021-03-01>).](https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_metric_alert_v2000001/providers/Microsoft.Web/serverfarms/plan1?api-version=2021-03-01%3E).)
E               Matchers succeeded : ['method', 'scheme', 'host', 'port', 'path']
E               Matchers failed :
E               _custom_request_query_matcher - assertion failure :
E               None

@StrawnSC Some failed tests are caused by the inconsistent api-versions in the yaml file due to the upgrade of `Microsoft.Web/serverfarms` api-version. Please re-record these failed tests in live mode and then upload yaml files.

File: src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:53]: @StrawnSC have you check an API directly using ARM to check these APIs infact don't support this API version or is it an SDK issue. @calvinsID FYI. Please inform Calvin & Gargi about this, so they can work on the required fixes.

Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:53]: yes, the issue occurs even without using the SDK. I'll loop in Calvin and Gargi on this

Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/appservice_domains.py:53]: @zhoxing-ms the new API version is getting released this weekend and I plan on reversing this particular change on Monday (so I marked the PR as do-not-merge). Other than that, the PR is ready to be reviewed

**Comment by StrawnSC** [comment]
Just added two minor new features/updates to this:

[App Service] az functionapp create: Add new --https-only parameter
[App Service] az functionapp/webapp create: Allow vnet integration for basic and elastic premium SKUs

**Comment by panchagnula** [comment]
> az functionapp create: Add new --https-only parameter

LA needs to be updated as well along with function apps. Did we add tests for the new changes?

**Comment by panchagnula** [comment]
Also given this PR is not just an SDK update anymore - please update the title so that changes are reflected in history. Thanks!

**Comment by StrawnSC** [comment]
> > az functionapp create: Add new --https-only parameter
> 
> LA needs to be updated as well along with function apps. Did we add tests for the new changes?

@panchagnula, I was planning on adding LA as well -- I just reached out to Arjun to double that we should block `--https-only` if the user provides a value for `--consumption-plan-location` since the work item says it's for standard only. 

I added a new test for the functionapp https only change (`test_functionapp_https_only`)

**Comment by StrawnSC** [comment]
> Also given this PR is not just an SDK update anymore - please update the title so that changes are reflected in history. Thanks!

Sounds good. I added to the "history notes" section earlier, but I'll update the title as well

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:2674]: May I ask why `Microsoft.DomainRegistration` and `Microsoft.CertificateRegistration` don't support the new api-version as `ResourceType.MGMT_APPSERVICE`?
If the supported api-version between them and `ResourceType.MGMT_APPSERVICE` are independent of each other, I suggest setting the sub configurations for them separately in the `_shared.py`, just like https://github.com/Azure/azure-cli/blob/b4759064b7e38203ef9d67a581105206a17f513b/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L151-L164

Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: The App Service certs and domains back end team just released the new API version, so I just removed all the places where I was hardcoding the API version in the client

**Comment by StrawnSC** [comment]
@panchagnula @zhoxing-ms I removed the changes where I set the `api_version` manually since the back end has been updated. This is ready to be merged once approved

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py
Comment (panchagnula) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: > @StrawnSC May I ask why `Microsoft.DomainRegistration` and `Microsoft.CertificateRegistration` don't support the new api-version as `ResourceType.MGMT_APPSERVICE`? If the supported api-version between them and `ResourceType.MGMT_APPSERVICE` are independent of each other, I suggest setting the sub configurations for them separately in the `_shared.py`, just like
> 
> https://github.com/Azure/azure-cli/blob/b4759064b7e38203ef9d67a581105206a17f513b/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L151-L164

@zhoxing-ms this was a temporary change, while we were waiting on ARM manifest update for the new API version to complete across all locations & get the initial code review unblocked.

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:236]: should the return_label be TRUE here - since this change will impact webapp up where we default this to True?

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_validators.py:339]: Can we confirm with function apps team this list accurate?

**Comment by panchagnula** [comment]
@StrawnSC add/update tests for the non SDK specific changes here?
https_only - for web apps, function apps, LA
SKU update in function apps - one or more of the new SKU is now supported?

**Review by panchagnula — CHANGES_REQUESTED** [review]
Please see comments.

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py
Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/_validators.py:339]: We confirmed this on the original email thread with the functions team that prompted this change

**Comment by StrawnSC** [comment]
> @StrawnSC add/update tests for the non SDK specific changes here?
> https_only - for web apps, function apps, LA
> SKU update in function apps - one or more of the new SKU is now supported?

@panchagnula I added new tests for this already -- `test_functionapp_https_only` and `test_logicapp_https_only`

**Comment by panchagnula** [comment]
> > @StrawnSC add/update tests for the non SDK specific changes here?
> > https_only - for web apps, function apps, LA
> 
> @panchagnula I added new tests for this already -- `test_functionapp_https_only` and `test_logicapp_https_only`
> 
> > SKU update in function apps - one or more of the new SKU is now supported?
> 
> I'll wasn't testing the new SKUs yet though. I'll add that right now

Hmm don't see the test files updated - only see the recordings. 
image

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py
Comment (StrawnSC) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:207]: @panchagnula here's the functionapp https-only test

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py
Comment (StrawnSC) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:118]: @panchagnula here's the logicapp https-only test

**Comment by StrawnSC** [comment]
> > > @StrawnSC add/update tests for the non SDK specific changes here?
> > > https_only - for web apps, function apps, LA
> > 
> > 
> > @panchagnula I added new tests for this already -- `test_functionapp_https_only` and `test_logicapp_https_only`
> > > SKU update in function apps - one or more of the new SKU is now supported?
> > 
> > 
> > I'll wasn't testing the new SKUs yet though. I'll add that right now
> 
> Hmm don't see the test files updated - only see the recordings. image

It looks like the recordings folder is still collapsed. The test files are show in your screenshot (`test_functionapp_com..`, etc.). I rerecorded more than 3 yaml files

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py
Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:236]: I've tested this and `return_label=True` causes broken behavior when the user calls this as `--https-only false`. Having `return_label=True` results in the value being passed as a string (ie `"true"` or `"false"`) instead of a boolean value (`True` or `False`). This behavior is explained in the documentation in the screenshot below. The result is that if we had `return_label=True` and the user called the command with `--https-only false`, they would have a webapp with httpsOnly true. 

I just doubled check this and it doesn't impact webapps created with `az webapp up` -- they still are httpsOnly by default: 
Screen Shot 2022-07-25 at 12 48 11 PM



Documentation for the return_label argument: 
Screen Shot 2022-07-25 at 12 38 11 PM

**Review by panchagnula — APPROVED** [review]
Approved. Just fix the failing tests & we should be good to merge as well.

**Comment by zhoxing-ms** [comment]
@StrawnSC Could you please resolve those test issues?

**Comment by StrawnSC** [comment]
> @StrawnSC Could you please resolve those test issues?

@zhoxing-ms sure, the CI tests are passing now

**Review by zhoxing-ms — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**
